fineract-provider/src/test/java/org/apache/fineract/infrastructure/core/LiquibaseDisabledScenario.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.BDDMockito.given;
+
+import com.zaxxer.hikari.HikariDataSource;
+import io.cucumber.java8.En;
+import org.apache.fineract.infrastructure.core.config.FineractProperties;
+import org.apache.fineract.infrastructure.core.service.migration.ExtendedSpringLiquibaseFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDataSourceFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDatabaseStateVerifier;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDatabaseUpgradeService;
+import org.apache.fineract.infrastructure.security.service.TenantDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class LiquibaseDisabledScenario implements En {
+
+    @Autowired
+    private TenantDetailsService tenantDetailsService;
+
+    @Autowired
+    private HikariDataSource tenantDataSource;
+
+    @Autowired
+    private FineractProperties fineractProperties;
+
+    @Autowired
+    private TenantDatabaseStateVerifier databaseStateVerifier;
+
+    @Autowired
+    private ExtendedSpringLiquibaseFactory liquibaseFactory;
+
+    @Autowired
+    private TenantDataSourceFactory tenantDataSourceFactory;
+
+    @Autowired
+    private TenantDatabaseUpgradeService underTest;
+
+    public LiquibaseDisabledScenario() {
+        Given("Liquibase is disabled", () -> {
+            given(databaseStateVerifier.isLiquibaseDisabled()).willReturn(true);
+        });
+
+        When("After properties set is executed", () -> {
+            underTest.afterPropertiesSet();
+        });
+
+        Then("The Liquibase configuration is unchanged", () -> {
+            assertTrue(databaseStateVerifier.isLiquibaseDisabled());
```

Comment:
Well, the verifications were meant to make sure that nothing gets interacted with in case Liquibase is disabled, like there should be no interaction with figuring out the current database state, preparing the Liquibase migrations, nothing.
The assertion here will always pass since it's a mock and has been initialized with a mock value of true.

Reply:
... right... the difference now is that the real deal (no mock) gets injected... so we can't use the mock verification that certain methods were called or not (or classes used).